.github/workflows/ci.yaml

```diff
@@ -17,37 +17,79 @@ jobs:
     steps:
       - uses: open-turo/actions-gha/test@v1
 
-  unit-test:
-    name: Unit test
+  integration-test:
+    name: Integration test
     runs-on: ubuntu-latest
     steps:
       - name: Checkout this repo
         uses: actions/checkout@v3
-      - name: Introduce problem that needs resolution
-        run: echo "fixme" > ./badshell.sh
-      - name: Create temporary pull request
-        id: cpr
-        uses: peter-evans/create-pull-request@v4
+      - name: Share the pull request number
+        run: echo "Working within PR ${{ github.event.pull_request.number }}"
+      - name: Search for comment within PR
+        id: before-add-fc
+        uses: peter-evans/find-comment@v2
         with:
-          title: Temp PR for unit testing purposes
-          base: main
-          branch: pr-for-unit-testing
-          branch-suffix: timestamp
+          issue-number: ${{ github.event.pull_request.number }}
+          comment-author: open-turo-bot
+          body-includes:
+            ${{ format('auto-gen for PR {0}',
+            github.event.pull_request.number)}}
           token: ${{ secrets.OPEN_TURO_GITHUB_TOKEN }}
-          draft: true
-      - name: Verify new PR created
-        run: |
-          echo "PR ${{ steps.cpr.outputs.pull-request-number }} created"
-          echo "PR operation '${{ steps.cpr.outputs.pull-request-operation}}'"
-      - name: Cleanup temporary pull request
-        if:
-          always() && ${{ steps.cpr.outputs.pull-request-operation}} ==
-          'created' && ${{ steps.cpr.outputs.pull-request-number }} != ''
-        uses: peter-evans/close-pull@v2
+      - name: Share comment id before adding
+        run:
+          echo "Found comment id '${{ steps.before-add-fc.outputs.comment-id
+          }}'"
+      - name: Verify comment is not already present on the PR
+        if: steps.before-add-fc.outputs.comment-id != ''
+        run: exit 1
```

Comment:
@dgschwindturo I think that when it fails, we should think about using the GHA commands to expose the error in a more user friendly way.
https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#example-creating-an-annotation-for-an-error

Reply:
Ah yes, excellent point.

Reply:
Think I have touched these three, previously quite silent errors, with a more informative error.